dashboard/app/models/lesson.rb

```diff
@@ -420,6 +421,24 @@ def update_activities(activities)
     script.fix_script_level_positions
   end
 
+  def update_objectives(objectives)
+    return unless objectives
+
+    previous_objectives = self.objectives
+
+    new_objectives = objectives.map do |objective|
+      persisted_objective = objective['id'].blank? ? Objective.new : Objective.find(objective['id'])
+      persisted_objective.description = objective['description']
+      persisted_objective.save!
+      persisted_objective
+    end
+
+    # As each objective can only be associated with one lesson, destory
+    # any removed objectives
+    (previous_objectives - new_objectives).each(&:destroy!)
+    self.objectives = new_objectives
```

Comment:
if you add the `dependent: :destroy` clause on the `has_many :objectives` association in the Lesson model, you should be able to skip tracking and destroying of `previous_objectives` here. This is what I would recommend since once we are in that state it becomes very difficult to accidentally leave stray objectives around.

Reply:
Oh perfect! I'll do that